yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,438 @@
+            f'https://{host}/api/videofile.php?video_id={video_id}&lifetime=8640000',
+            video_id, fatal=True, note='Downloading video file info', headers=headers)
+
+        formats = get_formats(host, video_file)
+
+        slug = f'{int(1E6 * (int(video_id) // 1E6))}/{1000 * (int(video_id) // 1000)}'
+        video_info = self._call_api(
+            f'https://{host}/api/json/video/86400/{slug}/{video_id}.json',
+            video_id, note='Downloading video info', headers=headers)
+
+        return {
+            'id': video_id,
+            'display_id': display_id,
+            'title': traverse_obj(video_info, ('video', 'title')),
+            'uploader': traverse_obj(video_info, ('video', 'user', 'username')),
+            'duration': parse_duration(traverse_obj(video_info, ('video', 'duration'))),
+            'view_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'viewed'))),
+            'like_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'likes'))),
+            'dislike_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'dislikes'))),
+            'age_limit': 18,
+            'formats': formats,
+        }
+
+
+class PornTopIE(InfoExtractor):
+    _VALID_URL = r'https?://(?P<host>(?:www\.)?porntop\.com)/video/(?P<id>\d+)(?:/(?P<display_id>[^/?]+))?'
+    _TESTS = [{
+        'url': 'https://porntop.com/video/101569/triple-threat-with-lia-lor-malena-morgan-and-dani-daniels/',
+        'md5': '612ba7b3cb99455b382972948e200b08',
+        'info_dict': {
+            'id': '101569',
+            'display_id': 'triple-threat-with-lia-lor-malena-morgan-and-dani-daniels',
+            'ext': 'mp4',
+            'title': 'Triple Threat With Lia Lor, Malena Morgan And Dani Daniels',
+            'description': 'md5:285357d9d3a00ce5acb29f39f826dbf6',
+            'uploader': 'PatrickBush',
+            'duration': 480,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id, host, display_id = self._match_valid_url(url).group('id', 'host', 'display_id')
+
+        webpage = self._download_webpage(url, video_id)
+
+        # find the VideoObject json object
+        video_obj_text = self._search_regex(
+            r'<script[^>]*>[^<]*schemaJson\s*=\s*(?P<json_ld>[^<]+VideoObject[^<]+)\s*;\s*var\s+script\s*=[^<]*</script>',
+            webpage, 'VideoObject', group='json_ld')
+        # there are javascript code within the declaration that would break json parsing
+        # the statistics values look like this: parseInt("2697"). remove the parseInt() function
+        video_obj_text = re.sub(r'parseInt\([^\d]+(\d+)[^\d]+\)', r'\1', video_obj_text)
+        # remove the function at "duration"
+        video_obj_text = re.sub(r'\(function\(duration\).*\(([^)]+)\)(\s*,\s*"thumbnailUrl")', r'\1\2', video_obj_text)
```

Comment:
this onw would still have to be manually done ig

Reply:
I have updated the regex to be more generic:

`video_obj_text = re.sub(r'\(function\([^)]*\)\s*\{[^}]*\}\s*\)\(\s*(["\'][^)]*["\'])\s*\)', r'\1', video_obj_text)`

i.e. reducing functions in the form `(function(xxx) { .... })("zzz")` to just `"zzz"`.

Thoughts?

Reply:
nice. So this could also go in js_to_json

Reply:
Ok. It's now moved to js_to_json.